### PR TITLE
[Port] Add required minimum version for azdata extension (#15010)

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Fixes https://github.com/microsoft/azuredatastudio/issues/15009

The Azdata extension doesn't check that the version currently installed supports the features it needs - so if someone has an older version of azdata installed then the extension and anything depending on it may break in various ways.

We should make it clear when a version is required and not allow the user to use any functionality until the version is updated.

### CODE REVIEW

**Review by chlafreniere — APPROVED** [review]

**Comment by chlafreniere** [comment]
@Charles-Gagnon I'm waiting on approval for the issue

**Review by kburtram — APPROVED** [review]

**Pull request merged**
